Mathlib/RingTheory/TwoSidedIdeal/Basic.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.Tactic.Abel
+import Mathlib.GroupTheory.GroupAction.SubMulAction
+import Mathlib.RingTheory.Congruence.Basic
+
+/-!
+# Two Sided Ideals
+
+In this file, for any `Ring R`, we reinterpret `I : RingCon R` as a two-sided-ideal of a ring.
+
+## Main definitions and results
+* `TwoSidedIdeal`: For any `NonUnitalNonAssocRing R`, `TwoSidedIdeal R` is the exactly `RingCon R`.
+* `TwoSidedIdeal.setLike`: Every `I : TwoSidedIdeal R` can be interpreted as a set of `R` where
+  `x ∈ I` if and only if `I x 0`.
+* `TwoSidedIdeal.addCommGroup`: Every `I : TwoSidedIdeal R` is an abelian group.
+
+
+-/
+
+open MulOpposite
+
+section definitions
+
+/--
+A two-sided ideal of a ring `R` is a subset of `R` that contains `0` and is closed under addition,
+negation, and absorbs multiplication on both sides.
+-/
+abbrev TwoSidedIdeal (R : Type*) [NonUnitalNonAssocRing R] := RingCon R
```

Comment:
Should it be `abbrev` or `def`, do we want Lean to unfold `TwoSidedIdeal`?

Reply:
Definitely `abbrev` if we're going to have it at all, but I'll wait for other reviewers to comment about whether we actually want the alternate spelling.

Reply:
I think having this alternate spelling is probably confusing; either we should have one name that functions both as a relation and a set, or we define
```lean
structure TwoSidedIdeal where
  toRingCon : RingCon R
```
to draw a clear distinction between the two APIs.

Reply:
I am okay both ways, not to use a structure with a single field could save some boilerplate probably. But I don’t really mind boilerplate code.

Reply:
Okay. I have switched to one-field-structure approach.